authz/src/main/java/io/grpc/authz/AuthorizationPolicyTranslator.java

```diff
@@ -0,0 +1,197 @@
+  }
+
+  private static Permission parseHeader(Map<String, ?> header) throws IllegalArgumentException {
+    String key = JsonUtil.getString(header, "key");
+    if (key == null) {
+      throw new IllegalArgumentException("\"key\" is absent.");
+    }
+    if (key.charAt(0) == ':'
+        || key.startsWith("grpc-")
+        || UNSUPPORTED_HEADERS.contains(key.toLowerCase())) {
+      throw new IllegalArgumentException(String.format("Unsupported \"key\" %s.", key));
+    }
+    List<String> valuesList = JsonUtil.getListOfStrings(header, "values");
+    if (valuesList == null || (valuesList != null && valuesList.isEmpty())) {
+      throw new IllegalArgumentException("\"values\" is absent or empty.");
+    }
+    Permission.Set.Builder orSet = Permission.Set.newBuilder();
+    for (String value: valuesList) {
+      orSet.addRules(
+          Permission.newBuilder().setHeader(
+            HeaderMatcher.newBuilder()
+            .setName(key)
+            .setStringMatch(getStringMatcher(value)).build()).build());     
+    }
+    return Permission.newBuilder().setOrRules(orSet.build()).build();
+  }
+
+  private static Permission parseRequest(Map<String, ?> request) throws IllegalArgumentException {
+    Permission.Set.Builder andSet = Permission.Set.newBuilder();
+    List<String> pathsList = JsonUtil.getListOfStrings(request, "paths"); 
+    if (pathsList != null) {
+      Permission.Set.Builder pathsSet = Permission.Set.newBuilder();
+      for (String path: pathsList) {           
+        pathsSet.addRules(
+            Permission.newBuilder().setUrlPath(
+              PathMatcher.newBuilder().setPath(
+                getStringMatcher(path)).build()).build());
+      }
+      if (pathsSet.getRulesCount() > 0) {
+        andSet.addRules(Permission.newBuilder().setOrRules(pathsSet.build()).build());
+      }
+    }
+    List<Map<String, ?>> headersList = JsonUtil.getListOfObjects(request, "headers"); 
+    if (headersList != null) {
+      Permission.Set.Builder headersSet = Permission.Set.newBuilder();
+      for (Map<String, ?> header: headersList) {           
+        headersSet.addRules(parseHeader(header));
+      }
+      if (headersSet.getRulesCount() > 0) {
+        andSet.addRules(Permission.newBuilder().setAndRules(headersSet.build()).build());
+      }
+    }
+    if (andSet.getRulesCount() == 0) {
+      return Permission.newBuilder().setAny(true).build();
+    }
+    return Permission.newBuilder().setAndRules(andSet.build()).build();
+  }
+
+  private static Map<String, Policy> parseRules(
+      List<Map<String, ?>> objects, String name) throws IllegalArgumentException {
+    Map<String, Policy> policies = new LinkedHashMap<String, Policy>();
+    for (Map<String, ?> object: objects) {
+      String policyName = JsonUtil.getString(object, "name");
+      if (policyName == null) {
+        throw new IllegalArgumentException("rule \"name\" is absent.");
```

Comment:
Is empty string okay? Ditto in translate()

Reply:
I agree we shouldn't allow empty strings. Added check.